monitoring/uss_qualifier/scenarios/astm/netrid/common_dictionary_evaluator.py

```
@@ -651,3 +661,74 @@ def _evaluate_operational_status(
                 key="skip_reason",
                 message=f"Unsupported version {self._rid_version}: skipping Operational Status evaluation",
             )
+
+    def _evaluate_ua_type(
+        self,
+        injected_val: Optional[str],
+        observed_val: Optional[str],
+        participants: List[ParticipantID],
+        query_timestamp: datetime.datetime,
+    ):
+        with self._test_scenario.check(
+            "UA type is present and consistent with injected one",
+            participants,
+        ) as check:
+            if observed_val is None:
+                check.record_failed(
+                    "UA type is missing",
+                    details="USS did not return any UA type",
+                    query_timestamps=[query_timestamp],
+                )
+            elif not observed_val:
+                check.record_failed(
+                    "UA type is empty",
+                    details="USS returned an empty UA type",
+                    query_timestamps=[query_timestamp],
+                )
+
+            try:
+                injected_val_enum = injection.UAType(injected_val)
+            except ValueError:
+                if observed_val != injection.UAType.NotDeclared:
+                    check.record_failed(
```

I think this should actually raise an exception -- the injection attempt should be rejected if uss_qualifier doesn't follow the injection API and I think [the injection API](https://github.com/interuss/automated_testing_interfaces/blob/417797f42e8ebd7cc1b71e82ec0fee0fe1e972a1/rid/v1/injection.yaml#L402) has the injected value as either omitted or a valid UAType, so there should be no way for us to reach this failure if uss_qualifier is working properly.

---

OK will change (and also take into account for future checks) that we should raise exceptions in cases where we injected invalid data.

---

Actually what I propose is to just adapt that check to fail when there is no injected value and that the observed value is different from `NotDeclared`. I'm not sure that we should start explicitly verifying the correctness of injected values (here we now don't need to try parsing `injected_val` anymore).

---

Thanks for the table @BenjaminPelletier.
I've integrated it in the repo (in this PR) slightly modified to give an overview on how those checks are implemented. And the new implementation should be correct wrt this.